PhpAmqpLib/Wire/IO/SocketIO.php

```diff
@@ -23,16 +36,20 @@ class SocketIO extends AbstractIO
     private $keepalive;
 
     /**
-     * @param string $host
-     * @param int $port
-     * @param float $timeout
-     * @param bool $keepalive
+     * @param string     $host
+     * @param int        $port
+     * @param float      $send_timeout
+     * @param bool       $keepalive
+     * @param float|null $read_timeout if null - defaults to $send_timeout
+     * @param int        $heartbeat
      */
-    public function __construct($host, $port, $timeout, $keepalive = false)
+    public function __construct($host, $port, $send_timeout, $keepalive = false, $read_timeout = null, $heartbeat)
```

Comment:
The third parameter `$timeout` was used as a read timeout and now it's changed to write timeout.
I believe you need to switch the parameter in the method's signature.